scylla/src/frame/value.rs

```diff
@@ -643,6 +696,27 @@ impl<T: Value> ValueList for Vec<T> {
     }
 }
 
+// Implement ValueList for maps, which serializes named values
+macro_rules! impl_value_list_for_map {
+    ($map_type:ident, $key_type:ty) => {
+        impl<T: Value> ValueList for $map_type<$key_type, T> {
+            fn serialized(&self) -> SerializedResult<'_> {
+                let mut result = SerializedValues::with_capacity(self.len());
```

Comment:
I just realized that __all__ invocations of `SerializedValues::with_capacity` in our driver are currently misusing this function. The capacity expected by this function refers to the total size of the serialized values, but everywhere we provide the value _count_. The preallocated vector will always be too small.

I guess the line you wrote can stay for now, but we should consider fixing it in the future. I created an issue: https://github.com/scylladb/scylla-rust-driver/issues/385